### PR TITLE
refactor: Refactored API Docs to reflect API changes

### PR DESCRIPTION
**Pull request by pedroCloudoki**

### CODE REVIEW

File: src/components/APICatalog/APICatalog.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ const APICatalog: React.FC<APICatalogProps> = ({ apisToDisplay, limit }) => {
                           <Chip
                             data-test-id={testIds.apiCardVersion}
                             color="secondary"
-                            label={apiDetails.apiVersion}
+                            label={apiDetails.apiVersion ? apiDetails.apiVersion : t("fallbacks.noVersion")}
```
Comment (a31859) [inline src/components/APICatalog/APICatalog.tsx:90]: Why not use an or? `apiDetails.apiVersion || t("fallbacks.noVersion")`

**Review by a31859 — CHANGES_REQUESTED** [review]

File: src/components/APICatalog/APICatalog.tsx (hunk 1 above)
Comment (pedroCloudoki) [reply src/components/APICatalog/APICatalog.tsx:90]: Fair comment, I changed this but I was supposed to go back on it, and I completely forgot. Thanks for letting me know.

File: src/pages/APIProductDetails/APIProductDetails.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ export const APIProductDetails: React.FC = () => {
 
       setCurrentAPIDetails({
         appsSubbed: appsSubbedToAPI,
-        documentation: currentAPI && currentAPI["docs"] ? currentAPI["docs"][0] : null,
+        documentation: currentAPI && currentAPI["apiDocs"] ? currentAPI["apiDocs"][0] : null,
```
Comment (a31859) [inline src/pages/APIProductDetails/APIProductDetails.tsx:97]: The `currentAPI["apiDocs"]` is not an array anymore indexing is not needed

File: src/util/mapAPIData.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  const mappedAPIData: APIDetails[] = apis.map((api) => {
+    hasVersions = !!api.apiVersions.length;
+
+    return {
+      /* An API that is 'live' (i.e., 'production accessible') is a) one that has versions, and b) has
+      its 'live' property set to 'true'. Ones that do NOT meet any of the above criteria are ones
+      that, presently, only have 'API Documentation' to show for it. */
+      access: hasVersions && api.apiVersions[0].live,
+      contract: hasVersions ? api.apiVersions[0].title : null,
+      description: api.apiDocs && api.apiDocs.productIntro || "",
```
Comment (a31859) [inline src/util/mapAPIData.ts:16]: Shouldn't the or be with `t("fallbacks.noDescription")`?

**Review by a31859 — CHANGES_REQUESTED** [review]

**Review by a31859 — APPROVED** [review]

**Pull request merged**